py/redrock/dataobj.py

```diff
@@ -118,8 +124,160 @@ def sharedmem_pack(self):
         if hasattr(self, 'Rcsr'):
             del self.Rcsr
 
+
+class SimpleSpectrum(object):
+
+    def __init__(self, wave, flux, ivar, R):
+        self.nwave=wave.size
+        self.wave=wave
+        self.flux=flux
+        self.ivar=ivar
+        self.R=R
+        self.Rcsr = self.R.tocsr()
+        self.wavehash = hash((len(wave), wave[0], wave[1], wave[-2], wave[-1]))
+
+
+class MPISharedTargets(object):
```

Comment:
Please add comments about how this class is implemented.  The creation of `_target_dictionary` from an existing set of targets and then repackaging those with `_get_targets` into another set of targets seems a maintenance risk and/or candidate for future cleanup refactor, so let's get some comments in here while the ideas are still fresh.